qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/vqe_ucc_factory.py

```diff
@@ -81,107 +86,167 @@ def __init__(
                 ansatz, the evaluated mean and the evaluated standard deviation.`
             kwargs: any additional keyword arguments will be passed on to the VQE.
         """
-        self.quantum_instance = quantum_instance
-        self.optimizer = optimizer
-        self.initial_point = initial_point
-        self.gradient = gradient
-        self.expectation = expectation
-        self.include_custom = include_custom
-        self.ansatz = ansatz
-        self.initial_state = initial_state
-        self.callback = callback
+        self._ansatz = ansatz
+        self._initial_state = initial_state
+
+        # Not needed
+        self._quantum_instance = quantum_instance
+        self._optimizer = optimizer
+        self._initial_point = initial_point
+        self._gradient = gradient
+        self._expectation = expectation
+        self._include_custom = include_custom
+        self._callback = callback
```

Comment:
The way that these are "deprecated" now is actually not correct because they do not still work any more (which means they are effectively already removed instead of "just" deprecated). I am, unfortunately, not sure what the best course of action is.

The only suggestion I have is to check during `get_solver` whether any of these internal attributes are `not None` (which they all (except `include_custom`) default to anyways) and if so, set the respective attribute of the solver.The downside of this is that a user needs to ensure to leave these attributes at `None` when switching to the correct new behavior, otherwise they will have there settings overwritten.